### PR TITLE
feat: partially support for local Node.js packages analysis

### PR DESCRIPTION
**Pull request by werifu**
I improved the packj to partially support the analysis for local Node.js packages.
Usage: replace original `npm:xxx` with `local_node:/path/to/xxx`, relative path is also ok.

It works, but just works.
**Partially support**, means that not all original features supported. Local packages do not have too many metadata, therefore it's hard to apply meta analysis...

image

captured project is from
https://github.com/microsoft/vscode-extension-samples/tree/main/custom-editor-sample

### CODE REVIEW

File: packj/audit/main.py [hunk 1]
```diff
@@ -516,7 +517,7 @@ def analyze_composition(pm_name, pkg_name, ver_str, filepath, risks, report):
 
 		if pm_name == 'pypi':
 			language=LanguageEnum.python
-		elif pm_name == 'npm':
+		elif pm_name == 'npm' or pm_name == 'local_node':
```
Comment (ashishbijlani) [inline packj/audit/main.py:520]: Minor nitpick: `local_nodejs` is more descriptive.

Comment (ashishbijlani) [inline packj/audit/main.py:520]: Minor nitpick: `local_nodejs` is more descriptive.

File: packj/audit/pm_proxy/local_node.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+# Partially analysis support for local Node.js packages. 
+#
+import json
+import logging
+import os
+
+
+from packj.audit.pm_proxy.pm_base import PackageManagerProxy
+
+class LocalNodeProxy(PackageManagerProxy):
```
Comment (ashishbijlani) [inline packj/audit/pm_proxy/local_node.py:12]: Same: `LocalNodeJSProxy`

File: packj/audit/pm_proxy/npmjs.py [hunk 3]
```diff
@@ -126,8 +136,7 @@ def get_release_history(self, pkg_name, pkg_info=None, max_num=-1):
 
 	def get_version(self, pkg_name, ver_str=None, pkg_info=None):
 		if not pkg_info:
-			pkg_info = self.get_metadata(pkg_name=pkg_name)
-		assert pkg_info and 'versions' in pkg_info, "package not found!"
```
Comment (ashishbijlani) [inline packj/audit/pm_proxy/npmjs.py:130]: Why remove the `assert` statement?

File: packj/audit/pm_util.py [hunk 4]
```diff
@@ -31,6 +31,8 @@ def get_pm_enum(pm_name):
 		return PackageManagerEnum.npmjs
 	elif pm_name == 'rubygems':
 		return PackageManagerEnum.rubygems
+	elif pm_name == 'local_node':
+		return PackageManagerEnum.local_node
```
Comment (ashishbijlani) [inline packj/audit/pm_util.py:35]: `local_nodejs`

File: packj/util/enum_util.py [hunk 5]
```diff
@@ -13,6 +13,9 @@ class PackageManagerEnum(Enum):
     packagist = 'packagist'
     dockerhub = 'dockerhub'
 
+    # Local package
+    local_node = 'local_node'
```
Comment (ashishbijlani) [inline packj/util/enum_util.py:17]: `local_nodejs`

**Review by ashishbijlani — CHANGES_REQUESTED** [review]

**Comment by werifu** [comment]
I have fixed all problems you mentioned above at [6cb5c5a](https://github.com/ossillate-inc/packj/pull/33/commits/6cb5c5a99db467498c4e26f61c503771f961235f) as well as some original minor bugs

File: packj/audit/pm_proxy/npmjs.py (hunk 3 above)
Comment (werifu) [reply packj/audit/pm_proxy/npmjs.py:130]: Deleted by mistake, I have recovered.

**Review by ashishbijlani — APPROVED** [review]

**Comment by Roviky** [comment]
i find that Local packages is hard to apply meta analysis too!!!!!!

**Pull request merged**